### ISSUE TITLE
Reward Normalization

### ISSUE THREAD

**Issue by AdamGleave** [HumanCompatibleAI/imitation#395]
It is often desirable to have the reward used for RL training be normalized. In particular, scaling up the reward has similar effects to scaling up the learning rate. Normalizing ensures a consistent reward scale across environments, avoiding the need to tweak learning rate per environment.

There are two instances where the ground truth reward is used:
  - `train_rl`, for training expert policies. This currently uses `VecNormalize` with `norm_obs=False`, and I think it's probably fine to continue with this approach.
  - `PreferenceComparisons` for `SyntheticGatherer`. This currently does not normalize rewards, but it probably should, as currently the scale of reward affects the stochasticity of synthetic comparisons.

Note if `VecNormalize` is used, then the normalization statistics will need to be stored (and restored) from checkpoints to restart training. But currently the code is not set up to continue training from checkpoints anyway (although inference from checkpoints is possible).

Additionally, all reward learning algorithms that train both a reward and policy jointly (`AdversarialTrainer`, the parent of `AIRL` and `GAIL`, and `PreferenceComparisons`) uses the output of a reward network. Currently `PreferenceComparisons` does not use a normalized reward at all; `AdversarialTrainer` uses `VecNormalize`

Rather than using `VecNormalize` I suspect it is better to either (a) regularize the output of the reward network (similar to https://github.com/HumanCompatibleAI/evaluating-rewards/blob/master/src/evaluating_rewards/rewards/preferences.py#L155) which will tend to keep the scale small and relatively consistent between runs; (b) explicitly normalize the output by adding a `RunningNorm` (or `BatchNorm`?) layer to the end.

**Comment by kmdanielduan** [comment]
I completed a simple implementation of that in #425.

**Comment by AdamGleave** [comment]
This issue has now been addressed, closing.
